lib/fauxpaas/artifact_builder.rb

```diff
@@ -17,24 +17,40 @@ def initialize(ref_repo:, factory:)
     # @param signature [ReleaseSignature]
     # @return The result of factory.new, pointing at the path
     def build(signature)
-      path = Pathname.new(Dir.mktmpdir)
-      add_references!(signature, path)
-      factory.new(path)
+      @artifact = factory.new
```

Comment:
This binds the builder to one artifact and gives it state -- not sure whether we want that or not.

If we do, I think we should accept the signature in the constructor and/or let it be set (builder style) and then build becomes no-arg.